### PR TITLE
chore(distributed_tracing): ensure last datadog parent id tag is always set

### PR DESCRIPTION
**Pull request by mabdinur**
Implements W3C Phase 3. The last datadog parent id is always propagated in a distributed trace (if ddtrace propagation style includes tracecontext). This change will reduce instances where flamegraphs contain missing spans.  

Note - This change does not require a release note. It is a follow up to a previous PR.

Follow up to: https://github.com/DataDog/dd-trace-py/commit/90a3e3fed47763abf64cb4c04c6a32f5c64076cb#diff-18af2a7682cf3014ab4ae236fc54d2454bbf5293e81e098b1d22d39bdb61012d
 
## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by brettlangdon** [comment]
> Note - This change does not require a release note. It is a follow up to a previous PR.

@mabdinur which PR?

**Comment by mabdinur** [comment]
> > Note - This change does not require a release note. It is a follow up to a previous PR.
> 
> @mabdinur which PR?

This change: https://github.com/DataDog/dd-trace-py/commit/90a3e3fed47763abf64cb4c04c6a32f5c64076cb#diff-18af2a7682cf3014ab4ae236fc54d2454bbf5293e81e098b1d22d39bdb61012d.

**Review by ZStriker19 — APPROVED** [review]

File: ddtrace/propagation/http.py [hunk 1]
```diff
@@ -921,6 +921,10 @@ def _resolve_contexts(contexts, styles_w_ctx, normalized_headers):
                 ts = _extract_header_value(_POSSIBLE_HTTP_HEADER_TRACESTATE, normalized_headers)
                 if ts:
                     primary_context._meta[W3C_TRACESTATE_KEY] = ts
+                # W3C Phase 3: Ensure the last datadog parent id is always set on the primary context
```
Comment (brettlangdon) [inline ddtrace/propagation/http.py:924]: I am not sure "Phase 3" has context here in the comment, does it?

**Review by brettlangdon — APPROVED** [review]
comment nit, otherwise lgtm

File: ddtrace/propagation/http.py (hunk 1 above)
Comment (mabdinur) [reply ddtrace/propagation/http.py:924]: Not really. It just makes sense to me. I will remove it

Comment (mabdinur) [inline ddtrace/propagation/http.py:924]:
```suggestion
                # Ensure the last datadog parent id is always set on the primary context
```

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**